### DISCUSSION TITLE
What is the recommended method for using the Experiment Grid API with custom algorithms?

### DISCUSSION THREAD

**Discussion by vrn-sn** [PKU-Alignment/omnisafe#302] [Q&A] [answered]
Hi all,

My team is implementing algorithms that don't come pre-implemented in OmniSafe, and we'd love to use the Experiment Grid API to compare them to baselines (and to each other). We can [add algorithms to the grid configuration](https://www.omnisafe.ai/en/latest/common/exp_grid.html#omnisafe.common.experiment_grid.ExperimentGrid.add) (via the `algo` parameter), but it appears that this takes a list of strings representing algorithms that are already implemented in OmniSafe (e.g. `PPO`, `PPOLag`, etc.).

What is the process for registering a novel algorithm so that its string representation can be used in Experiment Grid for comparison? We have tried [registering](https://github.com/PKU-Alignment/omnisafe/blob/main/omnisafe/algorithms/registry.py) our algorithms using `@registry.register` like in [the base PPO implementation](https://github.com/PKU-Alignment/omnisafe/blob/51a2692afab8afd0cbd2ab2e3d6aec0b79e9af01/omnisafe/algorithms/on_policy/base/ppo.py#L25), but we still run into issues since [this file](https://github.com/PKU-Alignment/omnisafe/blob/main/omnisafe/algorithms/__init__.py) seems to hardcode the list of available algorithms.

Of course, we can manually modify the OmniSafe code, but I would assume there is a better mechanism for this. Thanks!

**Comment by Gaiejj** [comment, marked as answer]
Sorry for the late reply. You can follow the registration of ``PPO-Lag``. Suppose your novel algorithm is ``ALGO`` and it is an on-policy algorithm in ``naive_lagrange``. You can register it in the ``__init__.py`` in ``naive_lagrange``, ``on-policy`` and ``algorithms`` folder. Just like what we have done on ``PPO-Lag``.

**Reply by vrn-sn** [reply]
Thanks!
